src/utilities.cc

```diff
@@ -248,7 +248,13 @@ pid_t GetTID() {
 #endif
   static bool lacks_gettid = false;
   if (!lacks_gettid) {
+#if (defined(OS_MACOSX) && defined(HAVE_PTHREAD_THREADID_NP))
+    uint64_t tid64;
+    pthread_threadid_np(NULL, &tid64);
```

Comment:
You should probably check the return value and assign -1 to `tid` in case of an error.